mlir/docs/Canonicalization.md

```diff
@@ -33,6 +33,11 @@ together.
 
 Some important things to think about w.r.t. canonicalization patterns:
 
+*   The goal of canonicalization is to make subsequent optimizations more
+    effective. Therefore, performance improvements are not necessary for
```

Comment:
What's the meaning of `more effective`? Either something has an effect or it doesn't. Do you mean `efficient`?

Also, I don't quite understand how that relates to the second sentence `Therefore, performance improvements are not necessary`.